iOS/Flows/Conversation/_New/ConversationViewController.swift

```diff
@@ -172,83 +223,83 @@ class ConversationViewController: FullScreenViewController,
         
         targetContentOffset.pointee = CGPoint(x: newXOffset, y: targetOffset.y)
     }
-}
 
-// MARK: - Message Loading and Updates
+    // MARK: - SwipeableInputAccessoryViewDelegate
 
-extension ConversationViewController {
-    
-    @MainActor
-    func initializeDataSource() async {
-        guard let controller = self.conversationController else { return }
+    /// The collection view's content offset at the first call to prepare for a swipe. Used to reset the the content offset after a swipe is cancelled.
+    private var initialContentOffset: CGPoint?
+    /// The last swipe position type that was registersed, if any.
+    private var lastPreparedPosition: SwipeableInputAccessoryView.SendPosition?
 
-        // Make sure messages are loaded before initializing the data.
-        if let mostRecentMessage = controller.messages.last {
-            try? await controller.loadPreviousMessages(before: mostRecentMessage.id)
+    func swipeableInputAccessoryDidBeginSwipe(_ view: SwipeableInputAccessoryView) {
+        self.initialContentOffset = self.collectionView.contentOffset
+        self.lastPreparedPosition = nil
+
+        self.collectionView.isUserInteractionEnabled = false
```

Comment:
When the swipe begins we remember the current content offset so we can reset the collection view if they cancel the swipe.

I also disable the collection view interaction to prevent the user from scrolling it while sending a message.